emily/lambda/src/operations/chainstate.rs

```diff
@@ -2,50 +2,148 @@
+            )
+        )?;
+
+    // Change behavior depending on how many entries were found.
+    match query_output.items().len() {
+
+        // No blocks found.
+        0 => Err(EmilyApiError::NotFound(
+            format!("No bock hash found for height {}", height)
+        )),
+
+        // A single block found; happy path.
+        1 => {
+            // Extract.
+            let item = query_output.items().first().unwrap();
+
+            // TODO:
+            // extract fields more safely.
+            let block_hash = item.get("BlockHash").unwrap().as_s().unwrap();
+            let block_height = item.get("BlockHeight").unwrap().as_n().unwrap();
+
+            // Package.
+            common::package_response(models::GetChainstateResponseContent {
+                block_hash: block_hash.clone(),
+                block_height: block_height.parse::<f64>().unwrap(),
+            }, 200)
+        },
+
+        // Multiple conflicting blocks found.
+        _ => {
+            let block_hashes = query_output.items()
+                .iter()
+                .map(|item| {
+                    item.get("BlockHash").unwrap().as_s().unwrap().clone()
+                })
+                .collect::<Vec<String>>();
+
+            Err(EmilyApiError::InternalService(
+                format!(
+                    "Multiple block hashes for height {}: [{}]",
+                    height,
+                    block_hashes.join(", "),
+                ).to_string()
+            ))
+        }
+    }
 }
 
 /// Handles the update of an existing chainstate
-pub fn handle_update_chainstate(
-    body:  Option<String>
+pub async fn handle_update_chainstate(
+    body: Option<String>,
+    context: &LambdaContext,
 ) -> Result<common::SimpleApiResponse, errors::EmilyApiError> {
-    common::deserialize_request::<models::UpdateChainstateRequestContent>(body).map(|request| models::UpdateChainstateResponseContent {
-            block_hash: request.block_hash,
-            block_height: request.block_height,
-        })
-    .and_then(|response| {
-        // Return 202 because this PUT operation won't be reflected in GET calls
-        // until the change is consistent in DynamoDB (<1 second).
-        common::package_response(response, 202)
-    })
+
+    let request: models::UpdateChainstateRequestContent =
+        common::deserialize_request::<models::UpdateChainstateRequestContent>(body)?;
+
+    let chainstate_entry: models::UpdateChainstateResponseContent = models::UpdateChainstateResponseContent {
+        block_hash: request.block_hash.clone(),
+        block_height: request.block_height,
+    };
+
+    let mut item: HashMap<String, AttributeValue> = HashMap::new();
+    item.insert("BlockHeight".to_string(), AttributeValue::N(request.block_height.to_string()));
+    item.insert("BlockHash".to_string(), AttributeValue::S(request.block_hash));
```

Comment:
Similarly, we could add a `IntoAttributeValueMap` trait to break out this boilerplate from the business logic.